### DISCUSSION TITLE
Can Hugging Face be used for deployment?

### DISCUSSION THREAD

**Discussion by xizi6689** [danny-avila/LibreChat#844] [Q&A] [answered]
### What features would you like to see added?

If possible, please make a tutorial, or give some useful tips!

### More details

By the way, Hugging Face supports docker.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by supercyx3** [comment]
+1

**Comment by danny-avila** [comment]
Yes: https://huggingface.co/spaces/LibreChat/LibreChat

But you will have to ask @fuegovic and @Berry-13 how they did it

**Comment by berry-13** [comment]
@wcgXi @supercyx3 you can take a look at the [LibreChat showcase channel](https://discord.com/channels/1086345563026489514/1136674701431820348). If you have any other questions you can ask them here or on Discord.

**Comment by fuegovic** [comment, marked as answer]
@wcgXi I documented the process:
https://docs.librechat.ai/deployment/huggingface.html

**Comment by peterhoang** [comment]
Just following up and wondering if anyone else is getting the same error:

```
> LibreChat@0.6.9 backend
> cross-env NODE_ENV=production node api/server/index.js

node:buffer:325
  throw new ERR_INVALID_ARG_TYPE(
  ^

TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
    at new NodeError (node:internal/errors:405:5)
    at Function.from (node:buffer:325:9)
    at Object.<anonymous> (/app/api/server/utils/crypto.js:4:20)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
    at Module.load (node:internal/modules/cjs/loader:1197:32)
    at Module._load (node:internal/modules/cjs/loader:1013:12)
    at Module.require (node:internal/modules/cjs/loader:1225:19)
    at require (node:internal/modules/helpers:177:18)
    at Object.<anonymous> (/app/api/server/utils/index.js:5:21) {
  code: 'ERR_INVALID_ARG_TYPE'
}

Node.js v18.19.0
npm notice 
npm notice New minor version of npm available! 10.2.3 -> 10.4.0
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v10.4.0>
npm notice Run `npm install -g npm@10.4.0` to update!
npm notice 
```

**Reply by fuegovic** [reply]
@peterhoang I'll update and test the template quickly.

**Reply by fuegovic** [reply]
@peterhoang I just updated the template, it's working fine for me
![image](https://github.com/danny-avila/LibreChat/assets/32828263/3d2ae0b5-25c2-4e30-909c-3310bdad43c1)

Make sure you're duplicating this space if you're following the guide: https://huggingface.co/spaces/LibreChat/template

**Comment by peterhoang** [comment]
Yes, I am duplicating the space. 

Unfortunately, I'm getting the same error. I noticed in your Dockerfile, you have a subdirectory, "/app". It's missing in the repo. Could that be why?

**Reply by fuegovic** [reply]
@peterhoang 
no, that's normal, it's a directory inside the docker image.

Looks like you didn't set all the required variables and secrets. 
there are info bubbles to help you in the process:
![image](https://github.com/danny-avila/LibreChat/assets/32828263/bdf2e282-fe51-4275-9c6e-4885a2abe791)
